### ISSUE TITLE
sql: panic in crdb_internal.create_test_objects when passed a large array

### ISSUE THREAD

**Issue by HonoreDB** [cockroachdb/cockroach#95081]
Random testing generated this SQL query:

```sql
SELECT array_to_string(IFNULL(array_agg(DISTINCT rowid), '{}'), '|') FROM seed WHERE CASE WHEN _bool THEN _jsonb ELSE json_remove_path(crdb_internal.generate_test_objects(_string::STRING, ARRAY[5390495026809626740]::INT8[])::JSONB::JSONB, (NULL::STRING[] || ARRAY['', '', '"']::STRING[])::STRING[]::STRING[])::JSONB END::JSONB IN (_jsonb, _jsonb, _jsonb, NULL)
```

Resulting in this error:

```
* ERROR: a panic has occurred!
* runtime error: makeslice: cap out of range
* (1) attached stack trace
*   -- stack trace:
*   | github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn.func1
*   | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:856
*   | [...repeated from below...]
* Wraps: (2) while executing: SELECT array_to_string(IFNULL(array_agg(DISTINCT _), '_'), '_') FROM _ WHERE CASE WHEN _ THEN _ ELSE json_remove_path(crdb_internal.generate_test_objects(_::STRING, ARRAY[_]::INT8[])::JSONB::JSONB, (_::STRING[] || ARRAY[_, _, __more1_10__]::STRING[])::STRING[]::STRING[])::JSONB END::JSONB IN (_, _, _, _)
* Wraps: (3) attached stack trace
*   -- stack trace:
... snipped
*   | runtime.panicmakeslicecap
*   | 	GOROOT/src/runtime/slice.go:33
*   | runtime.makeslice
*   | 	GOROOT/src/runtime/slice.go:100
*   | github.com/cockroachdb/cockroach/pkg/sql/catalog/randgen.(*testSchemaGenerator).selectNamesAndTemplates
*   | 	github.com/cockroachdb/cockroach/pkg/sql/catalog/randgen/tables.go:98
```

I'm guessing that `ARRAY[5390495026809626740]` somehow overflowed into a negative slice capacity at pkg/sql/catalog/randgen/tables.go:98 .

Jira issue: CRDB-23313

**Comment by yuzefovich** [comment]
@knz will this be fixed by #94987?

**Comment by knz** [comment]
@yuzefovich yes! But I'm still waiting for a review. Do you want to review it?
